src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -62,24 +89,32 @@ def dict_to_array(d):
 
     host = settings.GEOP['host']
     port = settings.GEOP['port']
-    path = settings.GEOP['path']
-    request = settings.GEOP['request'].copy()
-    request['input']['geometry'] = polygons
-    conn = httplib.HTTPConnection("%s:%s" % (host, port))
+    url = "http://%s:%s/jobs/%s" % (host, port, job_id)
 
-    try:
-        conn.request('POST', path, json.dumps(request))
-        data = json.loads(conn.getresponse().read())
-        if data['status'] == 'OK':
-            retval = [dict_to_array(d) for d in data['result']]
-        else:
-            retval = [[]]
-    except Exception:
-        retval = [[]]
-    finally:
-        conn.close()
+    get = requests.get(url)
+    if get.ok:
+        data = get.json()
+    else:
+        raise Exception('Unable to communicate with SJS (bottom-half).')
 
-    return retval
+    if data['status'] == 'OK':
+        if requests.delete(url).ok:  # job complete, remove
+            return [dict_to_array(d) for d in data['result']]
+        else:
+            raise Exception('Job completed, unable to delete.')
+    elif data['status'] == 'RUNNING':
+        try:
+            retry()
+        except MaxRetriesExceededError, X:  # job took too long, terminate
+            if requests.delete(url).ok:
+                raise X
```

Comment:
We may want to make `X` into something like `Exception('Job timed out, job deleted')`.